### PR TITLE
aes-sha2 support

### PR DESCRIPTION
**Pull request by greghudson**
It looks like the aes-sha2 RFC will be published soon enough for us to implement it in 1.15.  IANA has not assigned enctype and checksum type numbers yet, but supposedly will very soon.  The implementation is a big diff, so I am putting up the PR now for review.  I will update it with the assigned numbers as soon as they become available.

### CODE REVIEW

**Comment by frozencemetery** [comment]
The first commit removes the build support of hash_{md4,md5,sha1}.c and replaces them with hash_evp.c, but does not actually delete these files.

**Comment by greghudson** [comment]
Good catch; fixed.

**Comment by frozencemetery** [comment]
Looks good now, thanks.

**Comment by greghudson** [comment]
Updated with the assigned numbers.  (They may not be 100% official, but they appear in the public IANA registries and are unlikely to change.)

File: src/lib/crypto/krb/etypes.c [hunk 1]
```diff
@@ -167,6 +167,27 @@ const struct krb5_keytypes krb5int_enctypes_list[] = {
       krb5int_dk_cmac_prf,
       CKSUMTYPE_CMAC_CAMELLIA256,
       0 /*flags */ },
+
+    { ENCTYPE_AES128_CTS_HMAC_SHA256_128,
+      "aes128-cts-hmac-sha256-128", { "aes128-cts-sha256" },
```
Comment (tlyu) [inline src/lib/crypto/krb/etypes.c]: I'm still looking over the rest of the changes, but I would like us to use shorter aliases than "aes128-cts-sha256"; maybe "aes128-sha2" and "aes256-sha2"?

Comment (greghudson) [inline src/lib/crypto/krb/etypes.c]: Done.  I also added (via a new commit) aes128-sha1 and aes256-sha1 aliases.

**Comment by greghudson** [comment]
I realized the documentation still referred to the old alias names, so I fixed that up.  The documentation commit also adds the aes256-sha1 and aes128-sha1 aliases to the table.

**Comment by greghudson** [comment]
Pushed to master as 6fd74a89ac6c2444a347a357fac51b3490467284 and preceding.

**Pull request closed without merging**